dango/testing/src/hyperlane.rs

```diff
@@ -60,55 +60,90 @@ where
     pub fn new(
         mut suite: TestSuite<DB, VM, PP, ID>,
         mut owner: O,
-        size: usize,
-        threshold: usize,
-        remote_domain: Domain,
-    ) -> (Self, Shared<O>) {
-        let mock_validator_set = MockValidatorSet::new_random(size);
-
-        let ism = suite
-            .query_app_config::<AppConfig>()
-            .should_succeed()
-            .addresses
-            .hyperlane
-            .ism;
-
-        // Set validators at the ISM.
-        suite
-            .execute(
-                &mut owner,
-                ism,
-                &isms::multisig::ExecuteMsg::SetValidators {
-                    domain: remote_domain,
-                    threshold: threshold as u32,
-                    validators: mock_validator_set.addresses.clone(),
-                },
-                Coins::new(),
-            )
-            .should_succeed();
+        val_sets: BTreeMap<Domain, (usize, usize)>,
+    ) -> (HyperlaneTestSuite<DB, VM, PP, ID, O>, Shared<O>) {
+        let val_sets = val_sets
+            .into_iter()
+            .map(|(domain, (size, threshold))| {
+                let mock_validator_set = MockValidatorSet::new_random(size);
+
+                let ism = suite
+                    .query_app_config::<AppConfig>()
+                    .should_succeed()
+                    .addresses
+                    .hyperlane
+                    .ism;
```

Comment:
Extract this out of the iterator. Otherwise you're querying this once for every valset